cmd/main.go

```diff
@@ -1,79 +1,201 @@
 // SPDX-License-Identifier: Apache-2.0
 // Copyright (c) 2022-2023 Intel Corporation, or its subsidiaries.
 // Copyright (c) 2022-2023 Dell Inc, or its subsidiaries.
+// Copyright (C) 2023 Nordix Foundation.
 
 // Package main is the main package of the application
 package main
 
 import (
 	"context"
-	"flag"
 	"fmt"
+	"io"
 	"log"
 	"net"
 	"net/http"
+	"os"
+	"path/filepath"
+	"strconv"
 	"time"
 
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
 	pc "github.com/opiproject/opi-api/inventory/v1/gen/go"
 	pe "github.com/opiproject/opi-api/network/evpn-gw/v1alpha1/gen/go"
 	"github.com/opiproject/opi-evpn-bridge/pkg/bridge"
+	"github.com/opiproject/opi-evpn-bridge/pkg/config"
+	"github.com/opiproject/opi-evpn-bridge/pkg/infradb"
+	"github.com/opiproject/opi-evpn-bridge/pkg/infradb/taskmanager"
 	"github.com/opiproject/opi-evpn-bridge/pkg/port"
 	"github.com/opiproject/opi-evpn-bridge/pkg/svi"
 	"github.com/opiproject/opi-evpn-bridge/pkg/utils"
 	"github.com/opiproject/opi-evpn-bridge/pkg/vrf"
 	"github.com/opiproject/opi-smbios-bridge/pkg/inventory"
 
-	"github.com/philippgille/gokv"
-	"github.com/philippgille/gokv/redis"
-
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/credentials/insecure"
 	"google.golang.org/grpc/reflection"
 
 	"github.com/grpc-ecosystem/go-grpc-middleware/v2/interceptors/logging"
 	"github.com/grpc-ecosystem/grpc-gateway/v2/runtime"
-
+	ci_linux "github.com/opiproject/opi-evpn-bridge/pkg/LinuxCIModule"
+	gen_linux "github.com/opiproject/opi-evpn-bridge/pkg/LinuxGeneralModule"
+	ipu_linux "github.com/opiproject/opi-evpn-bridge/pkg/LinuxVendorModule/ipu"
+	frr "github.com/opiproject/opi-evpn-bridge/pkg/frr"
 	"go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc"
 )
 
-func main() {
-	var grpcPort int
-	flag.IntVar(&grpcPort, "grpc_port", 50151, "The gRPC server port")
+const (
+	configFilePath = "./"
+)
+
+var rootCmd = &cobra.Command{
+	Use:   "opi-evpn-bridge",
+	Short: "evpn bridge",
+	Long:  "evpn bridge application",
+	PreRunE: func(_ *cobra.Command, _ []string) error {
+		return validateConfigs()
+	},
+	Run: func(_ *cobra.Command, _ []string) {
+
+		taskmanager.TaskMan.StartTaskManager()
 
-	var httpPort int
-	flag.IntVar(&httpPort, "http_port", 8082, "The HTTP server port")
+		err := infradb.NewInfraDB(config.GlobalConfig.DBAddress, config.GlobalConfig.Database)
+		if err != nil {
+			log.Println("error in creating db", err)
+		}
+		go runGatewayServer(config.GlobalConfig.GRPCPort, config.GlobalConfig.HTTPPort)
 
-	var tlsFiles string
-	flag.StringVar(&tlsFiles, "tls", "", "TLS files in server_cert:server_key:ca_cert format.")
+		defer func() {
+			if err := infradb.Close(); err != nil {
+				log.Fatal(err)
+			}
+		}()
 
-	var redisAddress string
-	flag.StringVar(&redisAddress, "redis_addr", "127.0.0.1:6379", "Redis address in ip_address:port format")
+		switch config.GlobalConfig.Buildenv {
+		case "ipu":
+			gen_linux.Init()
+			ipu_linux.Init()
+			frr.Init()
+
+		case "ci":
```

Comment:
Hello @artek-koltun I will add a readme in the docs folder the where I will describe the architecture and also how the different modes are used. I will also add some flow digramms there. I hope that will do the work. The reason that I add that in the docs folder and not in the main Readme is because I do not want to pollute that with too much information.

Hope that is alright

Reply:
Hello @artek-koltun I have added the markdowns with the architecture and flow chart examples in the docs folder. You can take a look if you want.